src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> records = new ArrayList<>();
+        StringBuilder report = new StringBuilder();
+        readFromFile(fromFileName, records, report);
+        writeToFile(toFileName, report);
+    }
+
+    private static void readFromFile(String fromFileName,
+                                     List<String> stringFromFile, StringBuilder report) {
```

Comment:
The method 'readFromFile' is both reading from the file and building the report, which violates the single responsibility principle. Consider separating the logic for parsing and summarizing the data from the file reading logic.